exe/main.cc

```diff
@@ -26,7 +23,17 @@ namespace HW = HorseWhisperer;
 namespace lth_file = leatherman::file_util;
 
 int startAgent(std::vector<std::string> arguments) {
-#ifndef _WIN32
+#ifdef _WIN32
+    try {
+        Util::daemonize();
+    } catch (const std::exception& e) {
+        LOG_ERROR("Failed to daemonize: %1%", e.what());
+        return 1;
+    } catch (...) {
+        LOG_ERROR("Failed to daemonize");
+        return 1;
+    }
+#else
```

Comment:
Perhaps we can move the platform branch to Util::daemonize() and use a specific error for failures.